pupepat/utils.py

```diff
@@ -68,14 +128,63 @@ def save_results(input_filename: str, best_fit_models: list,
             best_fit_model['filename'] = os.path.basename(input_filename)
             for keyword in header_keywords:
                 best_fit_model[keyword] = hdu[0].header[keyword]
-            logger.info(best_fit_model)
+            logger.debug(best_fit_model)
             output_table.add_row(best_fit_model)
 
     output_table.write(output_filename, format='ascii', overwrite=True)
     return output_table
 
 
+def source_is_valid(data, source):
+    """
+    Validate source as reasonable. Checks:
+    1. bad columns in image
+    2. source is in focus (not a pupil)
+    3. too close to edge of image
+    4. not circular enough
+
+    :param data: image data from which the source was extracted
+    :param s: SEP extracted source
+
+    :return Boolean is_valid
+    """
+    bad_column_max = config['source_filters']['bad_column_max']
+    bad_column_min = config['source_filters']['bad_column_min']
+    got_bad_columns = (((source['xmax'] - source['xmin']) < bad_column_min and (source['ymax'] - source['ymin']) > bad_column_max) or
+                       ((source['xmax'] - source['xmin']) > bad_column_max and (source['ymax'] - source['ymin']) < bad_column_min))
+
+    focus_scale_factor = config['source_filters']['focus_scale_factor']
+    background = np.median(data)
+    in_focus = np.abs(data[int(source['y']), int(source['x'])] - background) > focus_scale_factor * np.sqrt(background)
```

Comment:
This is just a single pixel. It would probably be better to take the median of a small box around the center.